apps/x509.c

```diff
@@ -916,8 +916,19 @@ static ASN1_INTEGER *x509_load_serial(const char *CAfile,
     BIGNUM *serial = NULL;
 
     if (serialfile == NULL) {
-        const char *p = strchr(CAfile, '.');
-        size_t len = p != NULL ? (size_t)(p - CAfile) : strlen(CAfile);
+        const char *p = strrchr(CAfile, '.');
+        size_t len;
+        /* Check if the dot is in the file name or in one of subdirectories name */
+        if (p != NULL) {
+# ifdef WINDOWS
```

Comment:
I'm not sure this is really the ultimate solution. I know that path names in VMS are very different, see for example [test/conf_include_test.c#L23](https://github.com/openssl/openssl/blob/master/test/conf_include_test.c#L23). @levitte, any comments?

Reply:
I did not find a function to deal with paths in OpenSSL source code. I'm aware about VMS but I think it's better than current state.

Reply:
I didn't find one either. But having raw string handling and \#ifdef's everywhere in the code where paths need to be treated does not feel like the right solution. So either we should implement proper path handling functions somewhere, or just stick with the simple fix #6566 and document that a file extension is required for the `CAfile` (which is normally the case, anyway).